Doc/library/os.path.rst

```diff
@@ -488,6 +488,41 @@ the :mod:`glob` module.)
       Accepts a :term:`path-like object`.
 
 
+.. function:: splitroot(path)
+
+   Split the pathname *path* into a triad ``(drive, root, tail)`` where:
+
+   1. *drive* is a device name, mount point or the empty string;
+   2. *root* is a sequence of separators following the drive or the empty string; and
+   3. *tail* is anything after the root.
+
+   On POSIX systems, *drive* is always empty. The *root* may be empty (if *path* is
+   relative), a single forward slash (if *path* is absolute), or two forward slashes
+   (implementation-defined per `IEEE Std 1003.1-2017; 4.13 Pathname Resolution
+   <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_13>`_.)
+   For example::
+
+      >>> splitroot('/etc/hosts')
+      ('', '/', 'etc/hosts')
+
+   On Windows, *drive* may be empty, a drive-letter name, a UNC share, or a device
+   name. The *root* may be empty, a forward slash, or a backward slash. For
+   example::
+
+      >>> splitroot('Windows/notepad')
+      ('', '', 'Windows/notepad')
```

Comment:
```suggestion
      >>> splitroot('Users/Barney')
      ('', '', 'Users/Barney')
      >>> splitroot('/Users/Barney')
      ('', '/', 'Users/Barney')
```

Reply:
This doesn't seem necessary to me -- I think it can be inferred from the example immediately below. We should keep the docs here relatively terse, in keeping with the rest of the `os.path` documentation.

Reply:
If I had to choose just two Windows examples, I'd choose the following:

```
      >>> splitroot('C:/Users/Barney')
      ('C:', '/', 'Users/Barney')
      >>> splitroot('//Server/Share/Users/Barney')
      ('//Server/Share', '/', 'Users/Barney')
```

With 3 examples, I'd add a "//?/UNC" device example:

```
      >>> splitroot('//?/UNC/Server/Share/Users/Barney')
      ('//?/UNC/Server/Share', '/', 'Users/Barney')
```

Reply:
I've opted for 2 examples in this case as I think the `//?/UNC` behaviour should be obvious enough if you're using those sorts of paths.